text/0000-ember-data-friendly-errors.md

```diff
@@ -0,0 +1,90 @@
+- Start Date: 2015-10-23
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Add more illustrative detail to the default Ember Data Adapter Errors.
+
+# Motivation
+
+With a production Ember project, it's common to have many errors of the form "Adapter Error",
+originating from deep in the Ember Data stack and carrying little context about what the
+original error cause was.
+
+The intent is to add the original request URL, the response code, and some payload information
+to the default Error message for `DS.AdapterError`s. From there Errors can be handled or
+tracked as normal.
+
+# Detailed design
+
+I've been using something similar to the following Adapter (`friendly-error-adapter.js`):
+
+```js
+import ActiveModelAdapter from 'active-model-adapter';
+
+import DS from 'ember-data';
+
+export default ActiveModelAdapter.extend({
+
+  ajax(url, method)  {
+    this.lastRequest = {
```

Comment:
Yeah, mostly a hack to avoid trying to drop too deeply into the `ember-data` code. Definitely better ways to do it!